rspace/src/main/scala/coop/rchain/rspace/RSpaceOps.scala

```diff
@@ -140,16 +141,20 @@ abstract class RSpaceOps[F[_]: Concurrent, C, P, E, A, R, K](
     }
 
   override def reset(root: Blake2b256Hash): F[Unit] =
-    store.withTxnF(store.createTxnWriteF()) { txn =>
-      store.withTrieTxn(txn) { trieTxn =>
-        store.trieStore.validateAndPutRoot(trieTxn, store.trieBranch, root)
-        val leaves = store.trieStore.getLeaves(trieTxn, root)
-        eventLog.update(kp(Seq.empty))
-        store.getAndClearTrieUpdates()
-        store.clear(txn)
-        restoreInstalls(txn)
-        store.bulkInsert(txn, leaves.map { case Leaf(k, v) => (k, v) })
-      }
+    store.withTxnFlatF(store.createTxnWriteF()) { txn =>
+      for {
+        leaves <- store
+                   .withTrieTxn(txn) { trieTxn =>
+                     store.trieStore.validateAndPutRoot(trieTxn, store.trieBranch, root)
+                     store.trieStore.getLeaves(trieTxn, root)
+                   }
+                   .pure[F]
```

Comment:
delay :) pure is almost always wrong if you can't tell it's a pure value right away